python/activity_stream/activity_stream.py

```diff
@@ -0,0 +1,552 @@
+        self.ui.note_widget.set_data_retriever(data_retriever)
+        
+    ############################################################################
+    # public interface
+        
+    def load_data(self, sg_entity_dict):
+        """
+        Reset the state of the widget and then load up the data
+        for a given entity.
+        
+        :param sg_entity_dict: Dictionary with keys type and id
+        """
+        self._bundle.log_debug("Setting up activity stream for entity %s" % sg_entity_dict)
+        # clean up everything first
+        self._clear()
+        
+        # change the state
+        self._sg_entity_dict = sg_entity_dict
+        self._entity_type = self._sg_entity_dict["type"]
+        self._entity_id = self._sg_entity_dict["id"]
+        
+        # tell our "new note" widget which entity it should link up against 
+        self.ui.note_widget.set_current_entity(self._entity_type, 
+                                               self._entity_id)
+
+        # to mimic the behavior in shotgun - which seems quite strange and
+        # inconsistent for users, we need to disable to note dialog for 
+        # these cases
+        if self._entity_type in ["ApiUser", "HumanUser", "ClientUser"]:
+            self.ui.note_widget.setVisible(False)
+        else:
+            self.ui.note_widget.setVisible(True)
+
+        # now load cached data for the given entity
+        self._bundle.log_debug("Setting up db manager....")
+        ids_to_process = self._data_manager.load_activity_data(self._entity_type, 
+                                                               self._entity_id,
+                                                               self.MAX_STREAM_LENGTH)
+
+        if len(ids_to_process) == 0:
+            # nothing cached - show spinner!
+            # NOTE!!!! - cannot use the actual spinning animation because
+            # this triggers the GIL bug where signals from threads
+            # will deadlock the GIL
+            self.__overlay.show_message("Loading Shotgun Data...")
+
+        all_reply_users = []
+        attachment_requests = []
+        
+        ###############################################################
+        # Phase 1 - render the UI.
+        
+        # before we begin widget operations, turn off visibility
+        # of the whole widget in order to avoid recomputes
+        self._bundle.log_debug("Start building widgets based on cached data...")
+        self.setVisible(False)
+        try:
+
+            # we are building the widgets bottom up.
+            # first of all, insert a widget that will expand so that 
+            # it consumes all unused space. This is to keep other
+            # widgets from growing when there are only a few widgets
+            # available in the scroll area.
+            self._bundle.log_debug("Adding expanding base widget...")            
+            
+            expanding_widget = QtGui.QLabel(self)
```

Comment:
I picked a label to make sure that it wouldn't affect the external boundaries. Mostly because I knew this would work and wouldn't require lots of tweaking. A better be safe than sorry approach I guess. Do you think it's terrible? Maybe I should have used `QSpacerItem` with policy preferred?